.circleci/config.yml

```diff
@@ -238,7 +238,7 @@ jobs:
 
       - run:
           name: Run tests
-          command: pytest -vvrfEsx --skip-formatting --cov=prefect --cov-report=xml:/tmp/workspace/coverage/python36-coverage.xml .
+          command: pytest -vvrfEsx --cov=prefect --cov-report=xml:/tmp/workspace/coverage/python36-coverage.xml .
```

Comment:
Because the formatting flag was also used on doc tests we probably want to keep it here so "test failure" means an actual unit test and not a doc test

Reply:
By doc tests, you mean the test in the `docs` directory?

Generally I run `py.test src/prefect`, rather than letting pytest find any files in the directory tree that start with `test_`. So running the docs tests would be `py.test docs` and running the code tests is `py.test src`. Seems cleaner than adding a new mark, but up to you.

Reply:
You could also have pytest ignore the docs folder for general `py.test` runs (so you can keep your workflow), and only a `py.test docs` command runs the docs tests.

Reply:
> By doc tests, you mean the test in the docs directory?

Yup exactly.

> So running the docs tests would be py.test docs and running the code tests is py.test src

That makes sense; the original idea is to make doc tests opt _out_ so that outside developers know that we run doc tests (otherwise they're likely to get surprised when CI jobs start failing).

I don't really have a strong opinion about which pattern to use, @joshmeek / @lauralorenz might have a stronger opinion.

Reply:
> That makes sense; the original idea is to make doc tests opt out so that outside developers know that we run doc tests (otherwise they're likely to get surprised when CI jobs start failing).

Because I'm so used to running tests as:

```shell
$ py.test tests/
```

I actually missed that tests were also stored in the `docs/` folder (since py.test won't look outside the directories you pass explicitly). Definitely just a preference/what you're used to thing, could go either way.

Reply:
No opinion

Reply:
FWIW I'm of the folk who just runs `pytest` on a new project and would have been included in the contributors who would see that docs test exist from that. The `skip-formatting` flag though, if it is in practice the same as `pytest tests/`, seems unnecessary to me (in other words `pytest tests/` seems like the more standard/easier opt-out). I think there may be some nuance that `pytest tests/` and `pytest --skip-formatting` isn't exactly the same if the `formatting` attribute isn't on every test in the docs directory but not sure it's worth distinguishing for us.

Reply:
Ok, following that I'm going to leave the config as is, but update the CI to explicitly state the directories. Unqualified `py.test` will run everything, specify `tests` or `docs` if you only want to run one.

Reply:
Cool cool - so sounds like we should remove the`--skip-formatting` flag and adjust the CI jobs (including appveyor) to not use the flag